src/hooks/BalanceSummary.ts

```diff
@@ -0,0 +1,140 @@
+   */
+  calculatedAvailableBalanceInSats: number | null // in sats
+  /**
+   * @description frozen or locked balance manually calculated
+   * @deprecated this value must be used till backend v0.9.7 is released, and then be removed.
+   */
+  calculatedFrozenOrLockedBalanceInSats: number | null // in sats
+}
+
+type AccountBalanceSummary = BalanceSummarySupport & {
+  accountIndex: number
+}
+
+type WalletBalanceSummary = BalanceSummarySupport & {
+  accountBalances: AccountBalanceSummary[] | null
+}
+
+export const parseTotalBalanceString = (rawTotalBalance: BalanceString): BalanceSummary => {
+  const indexOfFirstWhitespace = rawTotalBalance.indexOf(' ')
+
+  if (indexOfFirstWhitespace < 0) {
+    // backend server version <=v0.9.6 will have an invalid "available balance"
+    // as the available balance is not returned in the raw string
+    return {
+      totalBalance: rawTotalBalance,
+      availableBalanceDontUseYet: rawTotalBalance,
+    }
+  }
+
+  const indexOfOpenBracket = rawTotalBalance.indexOf('(')
+  const indexOfCloseBracket = rawTotalBalance.indexOf(')')
+  if (indexOfOpenBracket < indexOfFirstWhitespace || indexOfCloseBracket <= indexOfOpenBracket + 1) {
+    throw new Error('Unknown format of TotalBalanceString')
+  }
+
+  const availableBalance = rawTotalBalance.substring(0, indexOfFirstWhitespace)
+  const totalBalance = rawTotalBalance.substring(indexOfOpenBracket + 1, indexOfCloseBracket)
+
+  return {
+    totalBalance,
+    availableBalanceDontUseYet: availableBalance,
+  }
+}
+
+/**
+ * @deprecated this is necessary for backend version <= v0.9.6; remove afterwards
+ */
+const calculateFrozenOrLockedBalance = (accountNumber: number, utxos: Utxos) => {
+  const accountUtxos = utxos.filter((it) => it.mixdepth === accountNumber)
+  const frozenOrLockedUtxos = accountUtxos.filter((utxo) => utxo.frozen || utxo.locktime)
+  return frozenOrLockedUtxos.reduce((acc, utxo) => acc + utxo.value, 0)
+}
+
+const EMPTY_BALANCE_DETAILS = {
+  totalBalance: null,
+  availableBalanceDontUseYet: null,
+  accountBalances: null,
+  calculatedAvailableBalanceInSats: null,
+  calculatedFrozenOrLockedBalanceInSats: null,
+} as WalletBalanceSummary
+
+const useBalanceSummary = (): WalletBalanceSummary => {
+  const currentWalletInfo = useCurrentWalletInfo()
+
+  const balanceSummary = useMemo(() => {
+    if (!currentWalletInfo) {
+      return EMPTY_BALANCE_DETAILS
+    }
+
+    const rawTotalBalance = currentWalletInfo.data.display.walletinfo.total_balance
+    const accounts = currentWalletInfo.data.display.walletinfo.accounts
+    const utxos = currentWalletInfo.data.utxos.utxos
+
+    try {
+      const walletBalanceSummary = parseTotalBalanceString(rawTotalBalance)
+      const utxosByAccount = utxos.reduce((acc, utxo) => {
+        const key = `${utxo.mixdepth}`
+        acc[key] = acc[key] || []
+        acc[key].push(utxo)
+        return acc
+      }, {} as { [key: string]: Utxos })
```

Comment:
Do we need this (as `calculateFrozenOrLockedBalance` will already filter for UTXOs of the respective account anyway)?

Reply:
You are right in the sense that there is some redundancy. But it is rather the other way around, `calculateFrozenOrLockedBalance` does not need to filter anymore as it is already provided with filtered utxos.
Thank you – will simplify! :+1: